opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/HibernateCriteriaConverter.java

```diff
@@ -427,7 +433,28 @@ public void visitBetween(final BetweenRestriction restriction) {
 
         @Override
         public void visitSql(final SqlRestriction restriction) {
-            m_criterions.add(org.hibernate.criterion.Restrictions.sqlRestriction(restriction.getAttribute()));
+            if (restriction.getParameters() != null && restriction.getParameters().length > 0) {
+                // Map our {@link Type} enum values to {@link org.hibernate.type.Type} equivalents 
+                org.hibernate.type.Type[] types = Arrays.stream(restriction.getTypes()).map(t -> { 
+                    switch(t) {
+                    case FLOAT:
+                        return new FloatType();
+                    case INTEGER:
+                        return new IntegerType();
+                    case LONG:
+                        return new LongType();
+                    case STRING:
+                        return new StringType();
+                    case TIMESTAMP:
+                        return new TimestampType();
+                    default: 
+                        throw new UnsupportedOperationException("Unsupported type specified in SqlRestriction");
+                    }
+                }).collect(Collectors.toList()).toArray(new org.hibernate.type.Type[0]);
```

Comment:
We could use `restriction.getTypes().length` instead of `0` here to avoid re-allocating the array.